src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -324,45 +325,51 @@ def _validate_service_principal(client, sp_id):
 
 
 def _build_service_principal(client, name, url, client_secret):
-    sys.stdout.write('creating service principal')
+    view = progress.DeterminateStandardOut()
```

Comment:
You should not create view directly. Instead, you can get a progress controller and add messages to it. 
Example: https://github.com/Azure/azure-cli/blob/ce10f471882402bf92764fc8e36882e07c1287bd/src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/custom.py#L32